File: microservices-demo/movie-ui/src/app/carousel/carousel-item/movie.ts
/**
 * A movie model
 */
export class Movie {
  name: string;
  director: string;
  producer: string;
  releaseYear: string;
  imageUrl: string;
  id: string;

  constructor(movie: Movie) {
    this.name = movie.name;
    this.director = movie.director;
    this.producer = movie.producer;
    this.releaseYear = movie.releaseYear;
    this.imageUrl = movie.imageUrl;
    this.id = movie.id;
  }
}

File: microservices-demo/movie/src/main/java/com/example/movie/model/Movie.java
package com.example.movie.model;

import lombok.Builder;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "movie")
@Data
@Builder
public class Movie {
    @Id String id;
    String name;
    String director;
    String producer;
    String releaseYear;
}
